core/src/components/badge/badge.ionic.scss

```
@@ -18,6 +18,14 @@
   font-weight: globals.$ionic-font-weight-medium;
 }
 
+// Badge Shapes
+// --------------------------------------------------
+
+/* Soft Badge */
+:host(.badge-soft) {
+  @include border-radius(globals.$ionic-border-radius-100);
+}
```

```suggestion
:host(.badge-soft) {
  @include border-radius(globals.$ionic-border-radius-200);
}

:host(.badge-xxsmall.badge-soft),
:host(.badge-xsmall.badge-soft),
:host(.badge-small.badge-soft) {
  @include border-radius(globals.$ionic-border-radius-100);
}
```

According to the Figma, when badge size is `xxs`, `xs` or `sm` the border radius is `100`, otherwise if it is `md`, `lg`, or `xl` it is `200`.